src/city.py

```diff
@@ -63,13 +63,13 @@ def add_connection(self, new_city):
         self.connected_cities.append(new_city)
 
     # TODO redesign this method
-    def remove_all_cubes(self, colour):
-        if not self.cubes[colour]:
-            raise NoCityCubesException(self, colour)
-        dropped_cubes = self.cubes[colour]
-        self.cubes[colour] = 0
+    def nullify_infection_level(self, colour):
+        if not self.infection_levels[colour]:
+            raise NoDiseaseInCityException(self, colour)
+        level_reduction = self.infection_levels[colour]
+        self.infection_levels[colour] = 0
         logging.debug(
-            f'Removed all {colour} cubes from {self}')
+            f'Removed all levels of {colour} disease from {self}')
```

Comment:
I suggest "{colour} disease infection in {self} dropped to zero level" for accordance.